server-templates/api.mustache

```diff
@@ -105,7 +105,8 @@ public class {{classname}}  {
   @{{httpMethod}}{{#subresourceOperation}}
   @Path("{{{path}}}"){{/subresourceOperation}}{{#hasConsumes}}
   @Consumes({ {{#consumes}}"{{{mediaType}}}"{{^-last}}, {{/-last}}{{/consumes}} }){{/hasConsumes}}{{#hasProduces}}
-  @Produces({ {{#produces}}"{{{mediaType}}}"{{^-last}}, {{/-last}}{{/produces}} }){{/hasProduces}}
+  @Produces({ {{#produces}}"{{{mediaType}}}"{{^-last}}, {{/-last}}{{/produces}} }){{/hasProduces}}{{#hasAuthMethods}}
+  {{#authMethods}}{{#isOAuth}}@RolesAllowed("**"){{/isOAuth}}{{/authMethods}}{{/hasAuthMethods}}
```

Comment:
`@RolesAllowed("**")` LGTM, however, just want to mention that we could take full control of authentication and [deny all requests in the auth Mechanism](https://github.com/projectnessie/nessie/blob/main/servers/quarkus-authn/src/main/java/org/projectnessie/server/authn/NessieAuthMechanism.java#L134) (if we wanted). That would complicate the mechanisms a bit, but would provide extra certainly that no holes are left in the API.

Reply:
These are imho different things.

You can return a failed Uni holding an unauthorized error from an auth mechanism, but that means no other auth mechanism will be attempted. This may not be desired in Polaris, because imho we are heading towards having many auth mechanisms, so even if mechanism A cannot authenticate, maybe mechanism B can.

`@RolesAllowed` is a way to say: please authenticate the caller before invoking this endpoint. Since proactive auth is disabled, no authentication would happen without this annotation when the endpoint is invoked.

Reply:
Fair point.